### ISSUE TITLE
[FEA]

### ISSUE THREAD

**Issue by cocorosiekz** [NVIDIA/NVTabular#325]
**Is your feature request related to a problem? Please describe.**
In many companies' recsys business, their raw recommender training data(no matter offline storage or streaming data) is usually organized in a [SequenceExample](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/example/example.proto#L300) way, which means in one example(stands for one user request context) it keeps all items' feature sets and one single context feature set(usually the requesting user's feature and some context feature). They serialize, storage and transfer data in this way for better storage efficiency. And TensorFlow also offers methods like [parse_sequence_example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/example/example.proto#L300), [parse_single_sequence_example](https://www.tensorflow.org/api_docs/python/tf/io/parse_single_sequence_example) to deserialize, make copies of context feature and pack them with every item's feature into a single trainable example. Then every example is a complete one and can be shuffled with others. The total operation is like a flatMap.

**Describe the solution you'd like**
Current NVT supports Parquet, not TFRecord. I'm not sure how this SequenceExample like nested structure are being used in Parquet(how to serialize, deserialize?). But if Parquet supports this nested structure, maybe we can offer one of the below two ways to solve this problem.

1. Let user define the parsing function and pass it into a flatMap function, which outputs the tabular like data.
2. In a more strict way like tf.io.parse_sequence_example, let user define context feature name/type and item feauture name/type, then let NVT parse the data into a tabular like data.

**Describe alternatives you've considered**

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

**Comment by cocorosiekz** [comment]
@EvenOldridge As we discussed in the Merlin meeting.
